mbway/src/main/java/com/adyen/checkout/mbway/country/CountryAdapter.kt

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2020 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by josephj on 16/12/2020.
+ */
+
+package com.adyen.checkout.mbway.country
+
+import android.content.Context
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.BaseAdapter
+import com.adyen.checkout.mbway.ui.R
+
+class CountryAdapter(private val context: Context) : BaseAdapter() {
+
+    private val countries: MutableList<CountryModel> = mutableListOf()
+
+    fun setItems(countries: List<CountryModel>) {
+        this.countries.clear()
+        this.countries.addAll(countries)
+        notifyDataSetChanged()
+    }
+
+    fun getCountries(): List<CountryModel> = countries
+
+    override fun getDropDownView(position: Int, convertView: View?, parent: ViewGroup?): View {
+        val view: View
+        val viewHolder: CountryViewHolder
+        if (convertView == null) {
+            view = LayoutInflater.from(context).inflate(R.layout.country_dropdown_view, parent, false)
+            viewHolder = CountryViewHolder(view)
+            view.tag = viewHolder
+        } else {
+            view = convertView
+            viewHolder = view.tag as CountryViewHolder
+        }
+        viewHolder.bindItem(getItem(position))
+        return view
+    }
+
+    override fun getView(position: Int, convertView: View?, parent: ViewGroup?): View {
+        val view: View
+        val viewHolder: CountryViewHolder
+        if (convertView == null) {
+            view = LayoutInflater.from(context).inflate(R.layout.country_view, parent, false)
```

Comment:
Waiting for Design decision right? I thought the spinner was going to be on the left, but this is fine for now I guess.